core/src/main/java/org/apache/iceberg/stats/StatsFormat.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.stats;
+
+import io.airlift.compress.Compressor;
+import io.airlift.compress.zstd.ZstdCompressor;
+import io.airlift.compress.zstd.ZstdDecompressor;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+final class StatsFormat {
+  private StatsFormat() {
+  }
+
+  static final int CURRENT_FORMAT_VERSION = 1;
+
+  static final int MAGIC_AS_NUMBER_LE = new BigInteger(swap(getMagic())).intValueExact();
+
+  static final int SUPPORTED_FLAGS = 0b1;
+  static final int FLAG_COMPRESSED = 0b1;
+
+  static byte[] getMagic() {
+    return new byte[] {0x50, 0x46, 0x49, 0x53};
```

Comment:
Question: Does this particular magic bytes correlate with anything in particular or did we just decide on it?

Reply:
This is from the format spec https://github.com/apache/iceberg-docs/pull/69:

> `Magic` is four bytes 0x50, 0x46, 0x49, 0x53 (short for: Plain Format for
  Indices and Statistics),

Reply:
This always allocates bytes for the Magic byte[] array, Is this because we want to re-use the array we are allocating here? Just wondering if this could be a static final byte array since it get's used in some contexts (like counting length) which don't require the allocation.

Reply:
While it is mutable, it should be fine to rely on people not modifying it. This isn't public.

Reply:
I don't see a benefit of declaring it as a constant field, and i do feel strongly about sharing mutable state that's not supposed to be changed.